state/megawatcher.go

```diff
@@ -117,7 +117,9 @@ func getUnitPortRangesAndPorts(st *State, unitName string) ([]network.PortRange,
 	} else if err != nil {
 		return nil, nil, errors.Annotate(err, "failed to get unit port ranges")
 	}
-	var compatiblePorts []network.Port
+	// For backward compatibility, if there are no ports opened, return an
+	// empty slice rather than a nil slice.
+	compatiblePorts := make([]network.Port, 0)
```

Comment:
or, given that port ranges are usually single numbers in practice make([]network.Port, 0, len(portRanges))